test/handler.js

```diff
@@ -969,14 +970,15 @@ describe('handler', () => {
                     ],
                     handler: function (request, reply) {
 
-                        return reply('ok');
+                        return reply(request.pre.before instanceof Error ? 'had pre error' : 'ok');
```

Comment:
failAction = log or ignore should not "takeover" the request like the failAction = error does. If I reply the error from the handler we'll not see this distinction and that's why is expected the 200 response.

So I did it this way to see if the error is reaching the request.pre and at the same time leaving the 200 response intact.
I'll haply refactor this to make it more understandable. I just not sure how. Maybe leaving those test as I found them and creating new ones just for the assign behavior?